mlflow/models/utils.py

```diff
@@ -950,12 +974,45 @@ def _is_scalar(x):
             )
     if input_schema.is_tensor_spec():
         return _enforce_tensor_schema(pf_input, input_schema)
+    elif HAS_PYSPARK and isinstance(original_pf_input, SparkDataFrame):
+        return _enforce_pyspark_dataframe_schema(
+            original_pf_input, pf_input, input_schema, flavor=flavor
+        )
     elif input_schema.has_input_names():
         return _enforce_named_col_schema(pf_input, input_schema)
     else:
         return _enforce_unnamed_col_schema(pf_input, input_schema)
 
 
+def _enforce_pyspark_dataframe_schema(
```

Comment:
For this function, you're not modifying the original spark dataframe, so changes made by `_enforce_mlflow_datatype` will not be applied to the spark dataframe. If this is acceptable, please add docstring to explain it.